### ISSUE TITLE
Add a roadmap/TodoList for JP schema

### ISSUE THREAD

**Issue by InfinityLoop1308** [Wenti-D/Astralwelkin#1]
Hi. I am very happy to see someone is developing  a JP schema for Trime.  I'd like to help improve it to make it an alternative to [Mozc](https://github.com/google/mozc).
A roadmap/Todolist will be helpful as a way to promote communication, since I am not using JP IME very often.

**Comment by Wenti-D** [comment]
Hello! I'm glad that someone can help me handle with these. I added something like todolist or directions recently, maybe you can check it.
I'm wondering whether using non-ascii characters like kana as input code is possible. If so, JP schema developing will be easier.

**Comment by InfinityLoop1308** [comment]
> Hello! I'm glad that someone can help me handle with these. I added something like todolist or directions recently, maybe you can check it. I'm wondering whether using non-ascii characters like kana as input code is possible. If so, JP schema developing will be easier.

Possibly yes but can not confirm. You may try it

**Comment by InfinityLoop1308** [comment]
For your roadmap:
Theme: Can be solved using liquid keyboard.
Schema: hard to give a trivial solution.

> 回车直出平假名，而非内部编码；

If hana can be input directly, this should not be a problem.

>  增加符号词库（例如输入「ほし」会出现「☆」「★」等等）

No idea about where to find the dict.

> 扩展标点、符号输入（例如输入「=」会出现「≒」「≞」等等）；

Maybe putting them in the liquid keyboard is a better solution.

> 输入预测（例如输入「あそ」会给出「遊ぶ」「遊び」等等）。

Seems still an issue even for Chinese.

**Comment by Wenti-D** [comment]
Well, for

> 扩展标点、符号输入（例如输入「=」会出现「≒」「≞」等等）；

I'm inspired by other IMEs like GBoard and MS JP IME:
<img src="https://user-images.githubusercontent.com/73067578/175897722-d0a5dd1d-1128-412b-a224-9db31bc38304.png" width=400>
![image](https://user-images.githubusercontent.com/73067578/175897974-66ff7a90-8dc1-481f-b422-ea28652f7918.png)

For

> 回车直出平假名

Actually at first I tried kana-as-code method, writing something like:
```
# schema
speller:
  alphabet: 'あいうえおかきくけこがぎぐげごさしすせそざじずぜぞたちつてとだぢづでどなにぬねのはひふへほばびぶべぼぱぴぷぺぽまみむめもやゆよらりるれろわをんぁぃぅぇぉゃゅょー'

# dict
ア	あ
```
But it cannot work.

Maybe I should consider lua script?

**Comment by InfinityLoop1308** [comment]
> Well, for
> 
> > 扩展标点、符号输入（例如输入「=」会出现「≒」「≞」等等）；
> 
> I'm inspired by other IMEs like GBoard and MS JP IME: <img alt="" width="400" src="https://user-images.githubusercontent.com/73067578/175897722-d0a5dd1d-1128-412b-a224-9db31bc38304.png"> ![image](https://user-images.githubusercontent.com/73067578/175897974-66ff7a90-8dc1-481f-b422-ea28652f7918.png) 

If you would like to do it in this way, you can just add dicts. You may use LunaPinyin as a reference.

> For
> 
> > 回车直出平假名
> 
> Actually at first I tried kana-as-code method, writing something like:
> 
> ```
> # schema
> speller:
>   alphabet: 'あいうえおかきくけこがぎぐげごさしすせそざじずぜぞたちつてとだぢづでどなにぬねのはひふへほばびぶべぼぱぴぷぺぽまみむめもやゆよらりるれろわをんぁぃぅぇぉゃゅょー'
> 
> # dict
> ア	あ
> ```
> 
> But it cannot work.
> 
> Maybe I should consider lua script?

I think you need to change the keyboard theme as well.

**Comment by InfinityLoop1308** [comment]
BTW, I am very curious about where you find these dictionaries. Could you provide some information?

**Comment by Wenti-D** [comment]
> I think you need to change the keyboard theme as well.

True. But the test above is after I change my theme, something like:
`- {click: 'あ', label: あ}`

> BTW, I am very curious about where you find these dictionaries. Could you provide some information?

First I searched through GitHub and found [this page](https://github.com/gkovacs/rime-japanese), containing dicts now I use.
But it's out of date now, so I found newer versions on the web. All dict URLs are written in txts in `dict_source.7z`.
